### PR TITLE
load flowpart and bagpart resources by name

### PR DESCRIPTION
**Pull request by giannik**
To avoid loading resources multiple times with content item that has bagpart and flowpart.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Flows/Startup.cs [hunk 1]
```diff
@@ -45,6 +47,9 @@ public override void ConfigureServices(IServiceCollection services)
             services.AddContentPart<FlowMetadata>();
 
             services.AddDataMigration<Migrations>();
+
+            services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Flows/Startup.cs:52]:
```suggestion
```

File: src/OrchardCore.Modules/OrchardCore.Widgets/ResourceManifest.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+using Microsoft.Extensions.Options;
+using OrchardCore.ResourceManagement;
+
+namespace OrchardCore.Widgets
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Widgets/ResourceManifest.cs:4]: Please use file-scope namespaces. add `;` at the end of the line and i'll save indentations.

File: src/OrchardCore.Modules/OrchardCore.Widgets/ResourceManifest.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                .SetUrl("~/OrchardCore.Widgets/Scripts/widgetslist.edit.min.js","~/OrchardCore.Widgets/Scripts/widgetslist.edit.js");
+        }
+
+        public void Configure(ResourceManagementOptions options)
+        {
+            options.ResourceManifests.Add(_manifest);
+        }
+    }
+
+
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Widgets/ResourceManifest.cs:29]:
```suggestion
```

File: src/OrchardCore.Modules/OrchardCore.Flows/ResourceManifest.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                .SetUrl("~/OrchardCore.Flows/Scripts/flows.edit.min.js","~/OrchardCore.Flows/Scripts/flows.edit.js");
+        }
+
+        public void Configure(ResourceManagementOptions options)
+        {
+            options.ResourceManifests.Add(_manifest);
+        }
+    }
+
+
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Flows/ResourceManifest.cs:29]:
```suggestion
```

File: src/OrchardCore.Modules/OrchardCore.Flows/ResourceManifest.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+using Microsoft.Extensions.Options;
+using OrchardCore.ResourceManagement;
+
+namespace OrchardCore.Flows
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Flows/ResourceManifest.cs:4]: Please use file-scope namespaces. add ; at the end of the line and i'll save indentations.

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]
I am okay with this change. Alternatively, you can just add asp-name to each of the scripts/styles and that would also prevent the double injections.

**Comment by MikeAlhayek** [comment]
@giannik was this to fix #1590? If so, please add Fix #1590 in the PR's description.

**Comment by giannik** [comment]
> @giannik was this to fix #15950? If so, please add Fix #15950 in the PR's description.

@MikeAlhayek no it does'nt solve #15950. but this is required to have bagpart and flowpart on same content item.

**Comment by giannik** [comment]
> I am okay with this change. Alternatively, you can just add asp-name to each of the scripts/styles and that would also prevent the double injections.

sure , i just think this is more clean.

File: src/OrchardCore.Modules/OrchardCore.Flows/ResourceManifest.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
+using Microsoft.Extensions.Options;
+using OrchardCore.ResourceManagement;
+
+namespace OrchardCore.Flows;
+
+    public class ResourceManagementOptionsConfiguration : IConfigureOptions<ResourceManagementOptions>
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Flows/ResourceManifest.cs:6]: Well now you need to remove the extra indentation in the file.

File: src/OrchardCore.Modules/OrchardCore.Widgets/ResourceManifest.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
+using Microsoft.Extensions.Options;
+using OrchardCore.ResourceManagement;
+
+namespace OrchardCore.Widgets;
+
+    public class ResourceManagementOptionsConfiguration : IConfigureOptions<ResourceManagementOptions>
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Widgets/ResourceManifest.cs:6]: here too. every line have extra 4 spaces that should be removed

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

**Pull request merged**